### PR TITLE
Adding Pooling Algorithms 

### PR DESCRIPTION
**Pull request by Navaneeth-Sharma**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: computer_vision/pooling_functions.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
+# Source : https://computersciencewiki.org/index.php/Max-pooling_/_Pooling
+# Importing the libraries
+import numpy as np
+from PIL import Image
+
+
+# Maxpooling Function
+def maxpooling(arr:list, size:int, stride:int) -> list:
```
[the remark this answers is not in the document]
Comment (Navaneeth-Sharma) [reply computer_vision/pooling_functions.py:8]: How to provide doctest ?

[the remark this answers is not in the document]
Comment (Rohanrbharadwaj) [reply computer_vision/pooling_functions.py:8]: Doctests are provided in the docstrings of functions or classes. They include the test and expected result/exception. See [https://docs.python.org/3.10/library/doctest.html?](url)

[the remark this answers is not in the document]
Comment (Navaneeth-Sharma) [reply computer_vision/pooling_functions.py:8]: Hi, I have added the doctests. Thank you

**Comment by cclauss** [comment]
As discussed in CONTRIBUTING.md, please run the doctests on your own computer to ensure that they pass.
* python3 -m doctest -v computer_vision/pooling_functions.py

**Comment by Navaneeth-Sharma** [comment]
Hi, 
The tests ran successfully in the local. 
I got the idea where it is giving error, but in local machine it doesn't come. How to tackle? Please give some suggestions 
Thanks 
Screenshot (325)

**Comment by cclauss** [comment]
As discussed in CONTRIBUTING.md, please run psf/black to format your code.

File: computer_vision/pooling_functions.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    array([[ 99., 118., 123.],
+           [ 75.,  84., 112.],
+           [108., 116., 110.]])
+
+    Args:
+        arr: numpy array
+        size: size of pooling matrix
+        stride: the number of pixels shifts over the input matrix
+    Returns:
+        numpy array of avgpooled matrix
```
Comment (cclauss) [inline computer_vision/pooling_functions.py:84]: Please move this content before the doctests.

File: computer_vision/pooling_functions.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
+# Source : https://computersciencewiki.org/index.php/Max-pooling_/_Pooling
+# Importing the libraries
+import numpy as np
+from PIL import Image
+from doctest import testmod
```
Comment (cclauss) [inline computer_vision/pooling_functions.py:5]: Move this down below...
```suggestion
```

File: computer_vision/pooling_functions.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        # reset the column index to 0
+        j = 0
+        avg_mat_j = 0
+
+    return updated_arr
+
+
+# Main Function
+if __name__ == "__main__":
+    # test the functions
```
Comment (cclauss) [inline computer_vision/pooling_functions.py:122]:
```suggestion
    from doctest import testmod

```

File: computer_vision/pooling_functions.py [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    # Loading the image
+    image = Image.open("path_to_image")
+
+    # Converting the image to numpy array and maxpooling, displaying the result
+    # Ensure that the image is a square matrix
+
+    # Parameters
+    size = 3
+    stride = 2
+    Image.fromarray(maxpooling(np.array(image), size, stride)).show()
```
Comment (cclauss) [inline computer_vision/pooling_functions.py:134]:
```suggestion
    Image.fromarray(maxpooling(np.array(image), size=3, stride=2)).show()
```

File: computer_vision/pooling_functions.py [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+
+    # Converting the image to numpy array and averagepooling, displaying the result
+    # Ensure that the image is a square matrix
+
+    # Parameters
+    size = 3
+    stride = 2
+    Image.fromarray(avgpooling(np.array(image), size, stride)).show()
```
Comment (cclauss) [inline computer_vision/pooling_functions.py:141]:
```suggestion
    Image.fromarray(avgpooling(np.array(image), size=3, stride=2)).show()
```

**Comment by Navaneeth-Sharma** [comment]
Hi, Any other reformatting needs to be done?

File: computer_vision/pooling_functions.py [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+        if i + size > arr.shape[0]:
+            # if the end of the matrix is reached, break
+            break
+        while j < arr.shape[1]:
+            # if the end of the matrix is reached, break
+            if j + size > arr.shape[1]:
+                break
+            # compute the maximum of the pooling matrix
+            updated_arr[mat_i][mat_j] = np.max(arr[i:i + size, j:j + size])
+            # shift the pooling matrix by stride of column pixels
```
Comment (Rohanrbharadwaj) [inline computer_vision/pooling_functions.py:48]:
```suggestion
                break
            # compute the maximum of the pooling matrix
            updated_arr[mat_i][mat_j] = np.max(arr[i : i + size, j : j + size])
            # shift the pooling matrix by stride of column pixels
```

Comment (Rohanrbharadwaj) [reply computer_vision/pooling_functions.py:48]: Same on line 103

File: computer_vision/pooling_functions.py [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+
+    return updated_arr
+
+
+# Main Function
+if __name__ == "__main__":
+    from doctest import testmod
+
+    testmod(name='avgpooling', verbose=True)
+
```
Comment (Rohanrbharadwaj) [inline computer_vision/pooling_functions.py:123]:
```suggestion

    testmod(name="avgpooling", verbose=True)

```

Comment (Navaneeth-Sharma) [reply computer_vision/pooling_functions.py]: Done ! Thanks

**Review by cclauss — APPROVED** [review]
Thanks for doing this!

**Comment by Navaneeth-Sharma** [comment]
Thank you for accepting the PR

**Pull request merged**